File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/receiverQueryApi/model/CaptionDisplay.java
package org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model;

public class CaptionDisplay {
    public String msgType;
    public cta708 cta708;
    public imsc1 imsc1;

    public class cta708 {
        public String characterColor;
        public double characterOpacity;
        public int characterSize;
        public String fontStyle;
        public String backgroundColor;
        public int backgroundOpacity;
        public String characterEdge;
        public String characterEdgeColor;
        public String windowColor;
        public int windowOpacity;
    }

    public class imsc1 {
        public String region_textAlign;
        public String content_fontWeight;
    }
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/receiverQueryApi/model/CaptionDisplay.kt
package org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model

data class CaptionDisplay (
    var msgType: String? = null,
    var cta708: cta708? = null,
    var imsc1: imsc1? = null
)

data class cta708 (
    var characterColor: String? = null,
    var characterOpacity: Double = 0.0,
    var characterSize: Int = 0,
    var fontStyle: String? = null,
    var backgroundColor: String? = null,
    var backgroundOpacity: Int = 0,
    var characterEdge: String? = null,
    var characterEdgeColor: String? = null,
    var windowColor: String? = null,
    var windowOpacity: Int = 0
)

data class imsc1 (
    var region_textAlign: String? = null,
    var content_fontWeight: String? = null
)
